src/providers/JsonRpcProvider.ts

```diff
@@ -33,24 +42,40 @@ export class JsonRpcProvider {
       // "latest", "earliest", and "pending" require no manipulation
       rpcTimeFrame = timeFrame;
     }
-    const nodeResponse = (await post(
-      this._rpcUrl,
-      buildRPCPostBody('eth_getBlockByNumber', [
-        rpcTimeFrame,
-        returnTransactionObjects,
-      ]),
-    )) as RPCBlock;
+    const req = async (): Promise<RPCBlock> => {
+      return await post(
+        this._rpcUrl[this._rpcUrlCounter],
+        buildRPCPostBody('eth_getBlockByNumber', [
+          rpcTimeFrame,
+          returnTransactionObjects,
+        ]),
+      ).catch((e) => {
+        if (e.code === 'ENOTFOUND') {
```

Comment:
I was thinking about that but wasn't sure what the best approach would be. I'll dive into the `post` function more; wasn't sure of everything that depends on it. My first instinct was to create a new handler function, but I think it would've been impractical. My thought was that it would've had to pass in the `_rpcUrlCounter`, `get()`, and catch error, and then also return an incremented counter value